### PR TITLE
feature/FOUR-18100 STORY Reassign tasks I manage

### PR DESCRIPTION
**Pull request by gproly**
## Descripción

As a Process Manager, I want to reassign tasks so that I can ensure that tasks are processed by the most suitable reviewers.

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-18100

### CODE REVIEW

File: ProcessMaker/Http/Controllers/Api/UserController.php [hunk 1]
```diff
@@ -140,6 +140,78 @@ public function index(Request $request)
...
+     *     ),
+     *     @OA\Response(
+     *         response=200,
+     *         description="List of users with task counts",
+     *         @OA\JsonContent(
+     *             type="object",
+     *             @OA\Property(
+     *                 property="data",
+     *                 type="array",
+     *                 ref="#/components/schemas/users",
```
Comment (nolanpro) [inline ProcessMaker/Http/Controllers/Api/UserController.php:168]: This should be `@OA\Items(ref="#/components/schemas/users"),`. Try running `php artisan l5-swagger:generate` to verify its working

File: ProcessMaker/Http/Controllers/Api/UserController.php [hunk 2]
```diff
@@ -140,6 +140,78 @@ public function index(Request $request)
...
+     *                 type="object",
+     *                 ref="#/components/schemas/metadata",
+     *             ),
+     *         ),
+     *     ),
+     * )
+     */
+    public function getUsersTaskCount(Request $request)
+    {
+        if (!(Auth::user()->can('view-users') ||
```
Comment (nolanpro) [inline ProcessMaker/Http/Controllers/Api/UserController.php:181]: Please do this in the routes with `->middleware('can:view-users|create-processes|....`

File: ProcessMaker/Http/Controllers/Api/UserController.php [hunk 3]
```diff
@@ -140,6 +140,78 @@ public function index(Request $request)
...
+                    ->orWhere('firstname', 'like', $filter)
+                    ->orWhere('lastname', 'like', $filter);
+            });
+        }
+        
+        $query->where('status', 'ACTIVE');
+
+        $subQueryString = "SELECT COUNT(id) FROM process_request_tokens WHERE user_id=users.id AND status='ACTIVE' AND element_type='task'";
+        $query->selectRaw("*, ({$subQueryString}) AS count");
+        $query->groupBy('users.id');
```
Comment (nolanpro) [inline ProcessMaker/Http/Controllers/Api/UserController.php:204]: We should be able to use more query builder functions here. Please look at https://stackoverflow.com/questions/18533080/laravel-eloquent-groupby-and-also-return-count-of-each-group

**Comment by CarliPinell** [comment]
Reassignment counter is working fine.
image

**Review by CarliPinell — APPROVED** [review]

**Pull request merged**